src/components/Fieldset/Fieldset.tsx

```diff
@@ -1,70 +1,89 @@
 import clsx from 'clsx';
-import React, { ReactNode } from 'react';
+import React, { ElementType, ReactNode } from 'react';
 import styles from './Fieldset.module.css';
-import Legend from '../Legend';
 
-export interface Props {
+type FieldsetProps = {
   /**
-   * Child node(s) that can be nested inside component
+   * The contents of the fieldset. We suggest a Fieldset.Title followed by
+   * interactive elements.
+   *
+   * Should be wrapped in a fragment to allow our styling to control the spacing
+   * between elements.
    */
-  children?: ReactNode;
+  children: ReactNode;
   /**
-   * CSS class names that can be appended to the component.
+   * Additional classnames passed in for styling.
    */
   className?: string;
+};
+
+type LegendProps = {
   /**
-   * Toggles the visibility of the form control legend
-   */
-  hideLegend?: boolean;
-  /**
-   * HTML legend title text
+   * The legend of the fieldset.
    */
-  legend?: string;
+  children: ReactNode;
   /**
-   * Slot for node to appear directly after legend text. Typically used to include a Toolip
+   * Additional classnames passed in for styling.
    */
-  legendAfter?: ReactNode;
+  className?: string;
+};
+
+type ItemsProps = {
   /**
-   * String for the optional label. By default it is '(optional)'
+   * The content of the control elements in the fieldset.
    */
-  optionalLabel?: string;
+  children: ReactNode;
   /**
-   * Required is passed down to the legend to display "optional" or not
+   * Type of element the immediate wrapper around the contents should be.
+   * @default 'div'
    */
-  required?: boolean;
+  as?: ElementType;
   /**
-   * String for the required label to add additional information if needed.
+   * Additional classnames passed in for styling.
    */
-  requiredLabel?: string;
-}
+  className?: string;
+};
 
 /**
- * Primary UI component for user interaction
+ * Helper sub-component for styling the title at the top of the component.
  */
-export const Fieldset = ({
-  className,
-  legend,
-  legendAfter,
-  hideLegend,
-  required,
+const Legend: React.FC<LegendProps> = ({ children, className }) => {
+  const componentClassName = clsx(className, styles['legend']);
+  return <legend className={componentClassName}>{children}</legend>;
+};
+
+/**
+ * Helper sub-component for styling the control elements in the component.
+ */
+const Items: React.FC<ItemsProps> = ({
```

Comment:
let's remove FC syntax:
```
const Items = ({...}: ItemsProps) => {
```

and move this out to a `FieldsetItems` file

Reply:
Right, sounds good